test/swar/BasicOperations.cpp

```diff
@@ -121,33 +119,31 @@ static_assert(math::moduloPowerOfTwo<4096>(4097) == 1);
 
 using S = SWAR<8, u32>;
 using BS = BooleanSWAR<8, u32>;
-static_assert(subtractOneUnsafe(S{S::Literal,    {1, 3, 4, 8}}).value()
-                                  == S{S::Literal, {0, 2, 3, 7}}.value());
-
-static_assert(isPowerOfTwo(S{S::Literal,      {1, 3, 4, 8}}).value()
-                              == BS{BS::Literal, {T, F, T, T}}.value());
-
-static_assert(isPowerOfTwo(S{S::Literal,      {3, 7, 11, 101}}).value()
-                              == BS{BS::Literal, {F, F, F, F}}.value());
-
-static_assert(isPowerOfTwo(S{S::Literal,      {2, 64, 128, 7}}).value()
-                              == BS{BS::Literal, {T, T, T, 0}}.value());
-
+template <typename S, typename BS>
+constexpr auto powerOfTwoTest(
+        const typename S::type (&input)[S::Lanes],
+        const bool (&expected)[S::Lanes]) {
+    return isPowerOfTwo(S{S::Literal, input}).value() == BS{BS::Literal, expected}.value();
+}
+static_assert(powerOfTwoTest<S, BS>({1, 2, 3, 4}, {1, 1, 0, 1}));
+static_assert(powerOfTwoTest<S, BS>({2, 3, 64, 77}, {1, 0, 1, 0}));
+static_assert(powerOfTwoTest<S, BS>({3, 65, 128, 0}, {0, 0, 1, 1}));
```

Comment:
This much copy&pasting is not valuable.
Did you consider packing all of the redundancy into an `X` macro?